booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,716 @@
+      return !norNode.isMissingNode() && !norNode.asBoolean(false);
+    }
+    return false;
+  };
+
+  private static final Consumer<MultiAttributeValidator> ALL_REQ_EQUIP = (mav) -> mav.submitAllMatching("requestedEquipments.*");
+  private static final JsonContentCheck NOR_PLUS_ISO_CODE_IMPLIES_ACTIVE_REEFER = JsonAttribute.allIndividualMatchesMustBeValid(
+    "All requested Equipments where 'isNonOperatingReefer' is 'false' must have 'activeReeferSettings'",
+    ALL_REQ_EQUIP,
+    JsonAttribute.ifMatchedThen(
+      IS_ACTIVE_REEFER_SETTINGS_REQUIRED,
+      JsonAttribute.path("activeReeferSettings", JsonAttribute.matchedMustBePresent())
+    )
+  );
+
+  private static final JsonContentCheck ISO_EQUIPMENT_CODE_AND_NOR_CHECK = JsonAttribute.allIndividualMatchesMustBeValid(
+    "All requested Equipments where ISOEquipmentCode is reefer code must have 'isNonOperatingReefer' flag",
+    ALL_REQ_EQUIP,
+    JsonAttribute.ifMatchedThen(
+      IS_ISO_EQUIPMENT_CONTAINER_REEFER,
+      JsonAttribute.path("isNonOperatingReefer", JsonAttribute.matchedMustBePresent())
+    )
+  );
+
+  private static final JsonContentCheck IS_SHIPPER_OWNED_CONTAINER = JsonAttribute.customValidator(
+    "valid Requested Equipments",
+    (body) -> {
+      var issues = new LinkedHashSet<String>();
+      var requestedEquipments = body.path("requestedEquipments");
+      for (var rq: requestedEquipments) {
+        var isShipperOwned = rq.get("isShipperOwned").asBoolean(false);
+        var commodities = requestedEquipments.get("commodities");
+        if (isShipperOwned) {
+          for (var commodity : commodities) {;
+            var cargoGrossVolume = commodity.get("cargoGrossVolume");
+              var cargoGrossVolumeUnit = commodity.get("cargoGrossVolumeUnit");
+            if(cargoGrossVolume != null && (cargoGrossVolumeUnit  == null || cargoGrossVolumeUnit.isEmpty()) ) {
+              issues.add("the cargo gross volume unit must be provided");
+            }
+          }
+        }
+      }
+      return issues;
+    }
+  );
+
+  private static final JsonContentCheck UNIVERSAL_SERVICE_REFERENCE = JsonAttribute.customValidator(
+    "Conditional Universal Service Reference",
+    (body) -> {
+      var universalExportVoyageReference = body.path("universalExportVoyageReference");
+      var universalImportVoyageReference = body.path("universalImportVoyageReference");
+      var universalServiceReference = body.path("universalServiceReference");
+      var issues = new LinkedHashSet<String>();
+      if (JsonAttribute.isJsonNodePresent(universalExportVoyageReference) || JsonAttribute.isJsonNodePresent(universalImportVoyageReference) ) {
+        if (JsonAttribute.isJsonNodeAbsent(universalServiceReference) ){
+          issues.add("The universalServiceReference must be present as either universalExportVoyageReference or universalExportVoyageReference are present");
+        }
+      }
+      return issues;
```

Comment:
Nit / For next PR:

Use `return Set.of("message ...");` / `return Set.of()`  instead of doing a `var issues = new LinkedHashSet<String>();` for cases like this where there is a 0-1 issue case (the pattern is fine if it is 0-N or a more complex check).